ray-operator/controllers/ray/utils/httpproxy_httpclient.go

```diff
@@ -5,30 +5,45 @@ import (
 	"io"
 	"net/http"
 	"time"
+
+	ctrl "sigs.k8s.io/controller-runtime"
 )
 
 type RayHttpProxyClientInterface interface {
 	InitClient()
 	CheckHealth() error
-	SetHostIp(hostIp string, port int)
+	SetHostIp(hostIp, podNamespace, podName string, port int)
 }
 
-func GetRayHttpProxyClient() RayHttpProxyClientInterface {
-	return &RayHttpProxyClient{}
+func GetRayHttpProxyClientFunc(mgr ctrl.Manager, useProxy bool) func() RayHttpProxyClientInterface {
+	return func() RayHttpProxyClientInterface {
+		return &RayHttpProxyClient{
+			mgr:      mgr,
+			useProxy: useProxy,
+		}
+	}
 }
 
 type RayHttpProxyClient struct {
-	client       http.Client
+	client       *http.Client
 	httpProxyURL string
+
+	mgr      ctrl.Manager
+	useProxy bool
 }
 
 func (r *RayHttpProxyClient) InitClient() {
-	r.client = http.Client{
+	r.client = &http.Client{
```

Comment:
Is there any reason for this change?

Reply:
It's mainly because GetHTTPClient() returns a pointer